src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Field.cs

```diff
@@ -12,11 +10,65 @@ public override BoundNode VisitFieldAccess(BoundFieldAccess node)
+        ///
+        /// For instance, tuple.Item8
+        /// produces fieldAccess(field=Item1, receiver=fieldAccess(field=Rest, receiver=ValueTuple for tuple))
+        /// </summary>
+        private BoundExpression RewriteTupleFieldAccess(BoundFieldAccess node, BoundExpression rewrittenReceiver)
+        {
+            var tupleField = node.FieldSymbol as TupleFieldSymbol;
+            if (tupleField == null)
+            {
+                return MakeFieldAccess(node.Syntax, rewrittenReceiver, node.FieldSymbol, node.ConstantValue, node.ResultKind, node.Type, node);
+            }
+
+            TupleTypeSymbol tupleType = (TupleTypeSymbol)tupleField.ContainingSymbol;
+            NamedTypeSymbol underlyingTupleType = tupleType.UnderlyingTupleType;
+
+            int tupleRemainder;
+            int tupleChainLength = TupleTypeSymbol.NumberOfValueTuples(tupleType.Arity, out tupleRemainder);
+
+            int fieldRemainder;
+            int fieldChainLength = TupleTypeSymbol.NumberOfValueTuples(tupleField.Index + 1, out fieldRemainder);
+
+            BoundExpression currentField = rewrittenReceiver;
+            NamedTypeSymbol currentLinkType = underlyingTupleType;
+
+            for (int fieldLinkIndex = 0; fieldLinkIndex < fieldChainLength; fieldLinkIndex++)
+            {
+                if (fieldLinkIndex != 0)
+                {
+                    currentLinkType = TupleTypeSymbol.GetNestedTupleType(currentLinkType, 1);
+                }
+
+                int containingLinkSize = 8;
+                int indexInLink = 8;
+                if (fieldLinkIndex == fieldChainLength - 1)
+                {
+                    if (tupleChainLength == fieldChainLength)
+                    {
+                        containingLinkSize = tupleRemainder;
+                    }
+
+                    indexInLink = fieldRemainder;
+                }
+
+                WellKnownMember wellKnownTypeMember = TupleTypeSymbol.GetTupleTypeMember(containingLinkSize, indexInLink);
+                var linkField = (FieldSymbol)Binder.GetWellKnownTypeMember(_compilation, wellKnownTypeMember, _diagnostics, syntax: node.Syntax);
+
+                FieldSymbol nestedFieldSymbol = linkField?.AsMember(currentLinkType);
+
+                TypeSymbol nestedFieldType = currentLinkType.TypeArguments[indexInLink - 1];
+                currentField = new BoundFieldAccess(node.Syntax, currentField, nestedFieldSymbol, node.ConstantValue, node.ResultKind, nestedFieldType);
+            }
 
-            return MakeFieldAccess(node.Syntax, rewrittenReceiver, field, node.ConstantValue, node.ResultKind, node.Type, node);
+            var expandedNode = (BoundFieldAccess)currentField;
+            return MakeFieldAccess(expandedNode.Syntax, expandedNode.ReceiverOpt, expandedNode.FieldSymbol, expandedNode.ConstantValue, expandedNode.ResultKind, expandedNode.Type, expandedNode);
```

Comment:
Do we need to call `MakeFieldAccess` on the result? If so, should the `oldNodeOpt` parameter be omitted?